### PR TITLE
fix: mount component in [data-cy-root]

### LINKED ISSUE #24385: Angular CT does not mount component within div[data-cy-root] root element
### Current behavior

Angular components are mounted inside a generated top-level div that is outside of our control
Screen Shot 2022-10-25 at 10 58 47 AM

### Desired behavior

Angular components should be mounted inside of `<div data-cy-root></div>` so as to accommodate for custom style specificity and to conform with the rest of our mounting frameworks

### Test code to reproduce

Any Angular app with CT setup, check out the DOM inside the AUT iframe to verify once a component has been mounted.

### Cypress Version

10.10.0

### Other

The rendering of the Angular component is managed by `TestBed`, specifically the [DOMTestComponentRenderer](https://github.com/angular/angular/blob/main/packages/platform-browser-dynamic/testing/src/dom_test_component_renderer.ts). We could probably override this to always target the `data-cy-root` element.

### PR DESCRIPTION
**Pull request by ZachJW34**
- Closes #24385 

### Additional details
Tweaks the Injectable `TestComponentRenderer` to inject the mounted Angular component into the standard `[data-cy-root]` element. Preserves the `rootId` that was being attached so as to avoid a breaking change if someone was relying on this selector (plus internal Angular Testbed still relies on this rootId existing).

### Steps to test
Throw a `.only` on https://github.com/cypress-io/cypress/blob/4bf516aef3cc12741921186071840e5e4fc1286a/system-tests/test/component_testing_spec.ts#L122

You can limit the version iteration so as to only test a single Angular version. Once it passes and you want to see the app itself, open up the scaffolded system test with `yarn cypress:open --project <path-tol-scaffolded-system-test>`

### How has the user experience changed?
Screen Shot 2023-02-14 at 9 44 52 AM

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: npm/angular/src/mount.ts [hunk 1]
```diff
@@ -169,6 +171,22 @@ function bootstrapModule<T> (
   return testModuleMetaData
 }
 
+@Injectable()
+export class CypressTestComponentRenderer extends TestComponentRenderer {
+  override insertRootElement (rootElId: string) {
```
Comment (lmiller1990) [inline npm/angular/src/mount.ts:176]: Did not know about `override` 🤯

**Review by lmiller1990 — APPROVED** [review]

File: npm/angular/src/mount.ts (hunk 1 above)
Comment (ZachJW34) [reply npm/angular/src/mount.ts:176]: Didn't need to use it here, but I wanted to follow the same pattern Angular is using for the `TestComponentRenderer`. Doubt I'll ever use it personally!

**Review by jordanpowell88 — APPROVED** [review]
LGTM! Great job with this Zac

**Comment by lmiller1990** [comment]
Ahh no, we need the CHANGELOG entry 😓 

Added: [306db99](https://github.com/cypress-io/cypress/pull/25807/commits/306db99d2636004525ab98920b5da64f19840ec8)

**Comment by lmiller1990** [comment]
Merging - was green, just pushed a commit to fix the changelog lint.

**Pull request merged**